centreon/src/Core/ResourceAccess/Application/UseCase/AddRule/AddRule.php

```diff
@@ -166,28 +171,32 @@ private function isAuthorized(): bool
     /**
      * @param DatasetFilter $filter
      *
-     * @return DatasetFilter
+     * @return DatasetFilter[]
      */
-    private function findLastLevelOfFilter(DatasetFilter $filter): DatasetFilter
+    private function findApplicableFilters(DatasetFilter $filter): array
     {
-        $applicableFilter = null;
+        $lastLevelFilter = null;
+        $parentLastLevelFilter = null;
+
         // recursive method to find the last 'stage' of filter (descending filter)
-        $findApplicableFilter = function (DatasetFilter $filter) use (&$findApplicableFilter, &$applicableFilter): DatasetFilter {
+
+        $findApplicableFilters = function (DatasetFilter $filter) use (&$findApplicableFilters, &$lastLevelFilter, &$parentLastLevelFilter): array {
```

Comment:
Can you add PHPDoc with the types of the variables for readability ? As we can't typehint the variables in the use